### PR TITLE
dRep

### PR DESCRIPTION
**Pull request by jj-umn**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by jj-umn** [comment]
https://drep.readthedocs.io/en/latest/index.html

Work In Progress.   

dRep dereplicate includes checkm ( https://github.com/Ecogenomics/CheckM/wiki/Installation ), which has a config file in the checkm package library that must point to the root dir of the reference data.   

The only available reference data is at:  https://data.ace.uq.edu.au/public/CheckM_databases/checkm_data_2015_01_16.tar.gz  

That needs to be downloaded and uncompressed.  Then one needs to activate the drep conda package and run checkm setRoot <path of uncompressed ref data>  which will update the config in the codebase.  

I'm not sure how that should be handled for a galaxy installation.

**Comment by bernt-matthias** [comment]
Not sure if related at all. But there is some ongoing (actually stalled at the moment) work on checkm here https://github.com/bernt-matthias/mb-galaxy-tools/tree/master/tools/checkm

**Comment by bgruening** [comment]
@jj-umn we can use a conda activate script to automatically download this reference database (its smallish) and set the config file automatically either after installation, see here for example: https://github.com/bioconda/bioconda-recipes/blob/ce917a1b17e03d52a2cce634eb9f98bd098a201b/recipes/picrust2/post-link.sh

**Comment by jj-umn** [comment]
@bgruening @bernt-matthias Thanks, Bjoern.  I'll see if I can update the dRep conda recipe.  

Matthias, does that also help with you checkm tools.  Looks like you have put a lot of work into them.   I've been trying to get dRep included for a workflow for Magnus Øverlie Arntzen <magnus.arntzen@nmbu.no> and Francesco Delogu <delogu.francesco@nmbu.no>.  You might check with them if they could use checkm tools separately.

**Comment by jj-umn** [comment]
@bgruening https://quay.io/repository/biocontainers/drep was built before checkm update.  
Do I need to do a new bioconda build num for drep to get the build 1 for checkm?   Or is there some other way to update the docker container?  
The conda env works if done after the checkm update.

**Comment by bgruening** [comment]
@jj-umn can you approve and merge: https://github.com/bioconda/bioconda-recipes/pull/19959

File: tools/drep/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+owner: iuc
+categories: 
+- Metagenomics
+description: dRep compares and dereplicates genome sets
+long_description: dRep can rapidly and accurately compare 
+  a list of genomes in a pair-wise manner. 
+  This allows identification of groups of organisms that share 
+  similar DNA content in terms of Average Nucleotide Identity (ANI).
+
+homepage_url: 'https://github.com/MrOlm/drep'
```
Comment (bgruening) [inline tools/drep/.shed.yml]: that should be turned into a suite definition I guess

File: tools/drep/drep_compare.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    <outputs>
+        <expand macro="common_outputs" />
+    </outputs>
+    <tests>
+        <expand macro="test_defaults_log">
+            <has_text text="dRep compare finished" />
+        </expand>
+    </tests>
+    <help><![CDATA[
+usage: drep compare [-p PROCESSORS] [-d] [-h] [-ms MASH_SKETCH]
```
Comment (bgruening) [inline tools/drep/drep_compare.xml]: can we remove this and include some description?

File: tools/drep/drep_dereplicate.xml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        </collection>
+        <expand macro="drep_outputs" />
+    </outputs>
+    <tests>
+        <expand macro="test_defaults_log">
+            <has_text text="dRep dereplicate finished" />
+        </expand>
+    </tests>
+    <help><![CDATA[
+usage: drep dereplicate [-p PROCESSORS] [-d] [-h] [-l LENGTH]
```
Comment (bgruening) [inline tools/drep/drep_dereplicate.xml]: adding some help text and removing this here would be nice

**Comment by bgruening** [comment]
@jj-umn I get this error

```
Unexpected error: <class 'FileNotFoundError'>
Traceback (most recent call last):
  File "/home/bag/miniconda3/envs/__drep@2.4.0/bin/checkm", line 611, in <module>
    checkmParser.parseOptions(args)
  File "/home/bag/miniconda3/envs/__drep@2.4.0/lib/python3.6/site-packages/checkm/main.py", line 848, in parseOptions
    self.qa(options)
  File "/home/bag/miniconda3/envs/__drep@2.4.0/lib/python3.6/site-packages/checkm/main.py", line 364, in qa
    binIdToModels = markerSetParser.loadBinModels(hmmModelInfoFile)
  File "/home/bag/miniconda3/envs/__drep@2.4.0/lib/python3.6/site-packages/checkm/markerSets.py", line 537, in loadBinModels
    with gzip.open(filename, 'rb') as f:
  File "/home/bag/miniconda3/envs/__drep@2.4.0/lib/python3.6/gzip.py", line 53, in open
    binary_file = GzipFile(filename, gz_mode, compresslevel)
  File "/home/bag/miniconda3/envs/__drep@2.4.0/lib/python3.6/gzip.py", line 163, in __init__
    fileobj = self.myfileobj = builtins.open(filename, mode or 'rb')
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpXxLZZ3/job_working_directory/000/8/working/outdir/data/checkM/checkM_outdir/storage/checkm_hmm_info.pkl.gz'
bag@bag:~/projects/code/tools-iuc/tools/drep$ ll /tmp/tmpXxLZZ3/job_working_directory/000/8/working/outdir/data/checkM/checkM_outdir/
bins/       checkm.log  storage/    
bag@bag:~/projects/code/tools-iuc/tools/drep$ ll /tmp/tmpXxLZZ3/job_working_directory/000/8/working/outdir/data/checkM/checkM_outdir/
bins/       checkm.log  storage/    
bag@bag:~/projects/code/tools-iuc/tools/drep$ ll /tmp/tmpXxLZZ3/job_working_directory/000/8/working/outdir/data/checkM/checkM_outdir/storage/
insgesamt 8
drwxr-xr-x 3 bag bag  100 Feb 26 00:14 ./
drwxr-xr-x 4 bag bag  100 Feb 26 00:14 ../
-rw-r--r-- 1 bag bag 1308 Feb 26 00:14 bin_stats.tree.tsv
-rw-r--r-- 1 bag bag 2172 Feb 26 00:14 phylo_hmm_info.pkl.gz
drwxr-xr-x 2 bag bag  940 Feb 26 00:14 tree/

```

**Comment by jj-umn** [comment]
Thanks.  I'm trying to trace an execution locally to see how can happen.

**Comment by jj-umn** [comment]
@bgruening  Using the taxonomy_wf checkm method the test passed.   So the issue occurs when using the lineage_wf method, so I'll investigating that.  
I'll also try to improve the help section.

**Comment by bgruening** [comment]
Ok, thanks a lot!

**Comment by bgruening** [comment]
@bernt-matthias we just got a request for checkm. Is there anything we can help?

**Comment by jj-umn** [comment]
@bgruening @bernt-matthias   I was thinking of changing the default checkm method to taxonomy_wf rather than lineage_wf.   That option seems to work more reliably.   I haven't had any problem with lineage_wf on the MSI galaxy server, but I can't seem to get that to work in the test.  

I looked a little at creating a checkm tool, but hesitated when viewing the multitude of options.  I can start that effort again if needed.

**Comment by bernt-matthias** [comment]
Hi there. Current state is here https://github.com/bernt-matthias/mb-galaxy-tools/tree/master/tools/checkm. Largely based on https://github.com/hexylena/argparse2tool und a lot of bash magic. 

As far as I remember the *_wf tools worked quite well already. Maybe you can try it. I'm completely fine with porting this to IUC.

My plan is start working on argparse2tool soon in order to do the automatic conversion completely automatic (without all the bash magic).

**Comment by jj-umn** [comment]
@bernt-matthias Porting to IUC seems like a good option.  Thanks.

**Comment by bgruening** [comment]
@jj-umn thanks a lot. Can you please clean the help a little bit and then we are ready to go :)

File: tools/drep/.shed.yml (hunk 1 above)
Comment (bgruening) [reply tools/drep/.shed.yml:13]: This is still missing, sorry have overlooked it ...

**Comment by bgruening** [comment]
Sorry, overlooked the shed.yml changed. Can you please also remove the WIP tag if you think it is ready?

**Comment by jj-umn** [comment]
@bgruening  Removed the WIP label.   I think this is ready to merge.

**Pull request merged**